backend/compact-connect/lambdas/python/staff-user-pre-token/user_scopes.py

```diff
@@ -74,13 +79,17 @@ def _process_compact_permissions(self, compact_name, compact_permissions):
 
     def _process_jurisdiction_permissions(self, compact_name, jurisdiction_name, jurisdiction_actions):
         # Ensure included actions are limited to supported values
-        disallowed_actions = jurisdiction_actions - {'write', 'admin'}
+        disallowed_actions = jurisdiction_actions - {'write', 'admin', 'readPrivate'}
         if disallowed_actions:
             raise ValueError(
                 f'User {compact_name}/{jurisdiction_name} permissions include disallowed actions: '
                 f'{disallowed_actions}',
             )
         for action in jurisdiction_actions:
-            # Two levels of authz
-            self.add(f'{compact_name}/{action}')
-            self.add(f'{compact_name}/{jurisdiction_name}.{action}')
+            if action == 'readPrivate':
+                # This action only has one level of authz, since there is no external scope for it
+                self.add(f'{compact_name}/{jurisdiction_name}.readPrivate')
```

Comment:
Yes, but this is referring to the behavior of our other scopes, where if you are granted to fine-grain permission at the jurisdiction level, you are also granted the course-grain scope permission to grant you access to call the api itself. In the case of reading, all staff users are implicitly granted the 'readGeneral' scope, so we do not need to add the course-grain scope explicitly here. I'll improve this comment for clarity.

Reply:
Ah ok. I had to stare at this for a while to get that, but I think I understand now. Given what I think this all does, what do you think of simplifying this logic a bit as follows? I think some version of this combined with your comments might be a little easier to follow.

```python
self.add(f'{compact_name}/{jurisdiction_name}.{action}')

# Grant coarse-grain scope, which provides access to the API itself
# Since `readGeneral` is implicitly granted to all users, we do not need to grant a coarse-grain scope for the `readPrivate` action
if action != 'readPrivate':
    self.add(f'{compact_name}/{action}')
```